src/background/Wallet/Wallet.ts

```diff
@@ -1705,10 +1708,30 @@ export class Wallet {
     emitter.emit('cloudflareChallengeIssued');
   }
 
+  async analyticsIdSet({ context }: WalletMethodParams) {
+    this.verifyInternalOrigin(context);
+    emitter.emit('analyticsIdSet');
+  }
+
   async screenView({ context, params }: WalletMethodParams<ScreenViewParams>) {
     // NOTE: maybe consider adding a more generic method, e.g.:
     // walletPort.request('sendEvent', { event_name, params }).
     this.verifyInternalOrigin(context);
+    /**
+     * Wait for analyticsId to be set before sending screenView event.
+     * This can happen on the first launch or during the migration to the new analytics ID
+     */
+    const analyticsId = await BrowserStorage.get<string>(analyticsIdKey);
+    if (!analyticsId) {
+      await Promise.race([
+        wait(5000),
+        new Promise<void>((resolve) => {
+          emitter.on('analyticsIdSet', () => {
```

Comment:
`browser.storage.onChanged.addListener` can be used instead

Reply:
Thank you! This will be helpful!